src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpCertificateHelper.cs

```diff
@@ -74,5 +75,50 @@ internal static class CertificateHelper
                 }
```

Comment:
Line 72 is missing a Debug.Assert.
We should probably throw an `InternalException()` to signal other potential issues instead of continuing with a name mismatch error which could be considered as a less important issue?